### PR TITLE
Switch to jbuilder

### PR DESCRIPTION
**Pull request by samoht**
superseeds #839 /cc @avsm

### CODE REVIEW

**Comment by hannesm** [comment]
I tried to raise the question elsewhere, but didn't get any answer, so trying here again:
You rename `mirage_types.mli` to `mirage_types.ml`.  How does this interact with documentation generation?  Are the documentation comments of the `.ml` used when generating documentation?
Is there a difference (`diff -wur`!?) between the old way of generating documentation and the new way (and if so, is the diff ok or is there something missing)?

**Comment by samoht** [comment]
> How does this interact with documentation generation?

That doesn't change anything, `odoc` works as well as before -- the generated `index.html` page is identical.

**Comment by hannesm** [comment]
On 20/07/2017 11:26, Thomas Gazagnaire wrote:
> That doesn't change anything, `odoc` works as well as before -- the generated `index.html` page is identical.

Ok, so a `diff -urw` between old and new docs is empty?

**Comment by samoht** [comment]
> Ok, so a `diff -urw` between old and new docs is empty?

Yes, at least for `mirage_types.ml{i}`. I can test on something a bit more complex if you like.

**Comment by samoht** [comment]
Same for `mirage_types_lwt.ml{i}`

**Comment by hannesm** [comment]
@samoht thanks for investigating!

File: types/jbuild [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+   (public_name mirage-types)
+   (modules     (mirage_types))
+   (libraries   (mirage-device mirage-time mirage-random mirage-flow
+                 mirage-console mirage-block mirage-clock mirage-net
+                 mirage-kv mirage-fs mirage-channel
+                 mirage-protocols mirage-stack))))
+
+(library
+  ((name        mirage_types_lwt)
+   (public_name mirage-types-lwt)
```
Comment (rgrinberg) [inline types/jbuild]: Looks a bit fishy, shouldn't this have `(wrapped false)`? (Same for `mirage_types`).

Comment (samoht) [inline types/jbuild]: In practice that doesn't matter as there is only one module with that name, but it's probably saner to add `(wrapped false)`. Thanks for noticing :-)

**Pull request merged**